File: primes.pl
#!/usr/bin/perl

$n = int($ARGV[0]) or die 'missing parameter maximum number';
@composite;
for $i (2 .. int(sqrt($n))) {
 if (!$composite[$i]) {
  for ($j = $i*$i; $j <= $n; $j += $i) {
	$composite[$j] = 1;
  }
 }
}

$numprimes=0;
for my $i (2 .. $n) {
	$composite[$i] || $numprimes++;
}

print $numprimes."\n";

File: primes.php
<?php
if (empty($argv[1])) die('missing parameter limit');
$limit = (int) $argv[1];

$nums=[];
$sqrtlimit = (int)sqrt($limit);
for ($n = 2; $n <= $sqrtlimit; $n++)
	if (!isset($nums[$n]))
		for ($i = $n*$n; $i <= $limit; $i += $n)
			$nums[$i] = 1;

echo $limit-1-array_sum($nums)."\n";
